File: java/pe020.java
//file: pe020.java

import java.math.BigInteger;

public class pe020 {
  public static void main( String[] args ) {
    final int last = 100;
    BigInteger factorial = BigInteger.ONE;
    for (int i=0; i<last; i++)
      factorial = factorial.multiply(BigInteger.valueOf(i+1));

    String fstring = factorial.toString();
    int fsum = 0;
    for (int i = 0; i < fstring.length(); i++)
      fsum += Character.getNumericValue(fstring.charAt(i));

    System.out.println(fsum);
  }
}

File: typescript/pe020.ts
function pe020(last20:number): number {
  let factorial = 1n

  for(let i=0; i < last20; i++)
    factorial *= BigInt(i+1)

  let fstring = factorial.toString()
  let flen = fstring.length

  let fsum = 0
  for(let i=0; i < flen; i++)
    fsum += Number(fstring[i])
  return fsum
}

let last20 = 100
console.log(pe020(last20))
